rosbag2_storage/include/rosbag2_storage/storage_interfaces/read_only_interface.hpp

```diff
@@ -45,6 +47,8 @@ class ROSBAG2_STORAGE_PUBLIC ReadOnlyInterface
   virtual void set_filter(const StorageFilter & storage_filter) = 0;
 
   virtual void reset_filter() = 0;
+
+  virtual void seek(const rcutils_time_point_value_t & timestamp) = 0;
```

Comment:
I know the rest of this interface isn't documented yet - but can you document this new signature? I'd like to see some language like "next message from `read_next` will be greater than or equal to `timestamp`". Also some note about the bounds behavior, such as does it automatically clamp the value to `[start, end]`?

Reply:
Second comment - I think this should probably `rclcpp::Time` instead of an `rcutils` type. Open to counterpoint, though

Reply:
@emersonknapp We have defined `virtual void jump(rcutils_time_point_value_t time_point) = 0;` API with rcutils timestamp in `PlayerClock` class and I am going to operate with it on upper `rosbag2_transport` layer as well. 
It would be good to have consistency in types in all API levels to avoid extra conversions or checks.

Reply:
I was just keeping it consistent with the type returned by `read_next()`, if `PlayerClock::jump` uses same type, then i vote not changing it for now.

Reply:
Good counterpoint!

Reply:
If I may be so bold as to comment on an issue I'm not actively reviewing, perhaps we should create a separate issue where we discuss the merits of both `rclcpp::Time` and `rcutils`, and use that to decide which way the package as a whole should go?

My vote is also for not changing the API yet. But that's mostly because I like having one PR per one issue, and that feels a bit beyond scope here.